pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -215,6 +215,23 @@ func isCreated(pod *v1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+// isStatefullyStuck returns true if a pod in a stateful set is stuck
+// due to a previously bad roll-out. We can detect this by checking all of:
+// 1) The pod is in a pending state
+// 2) The pod is at a different revision than the update revision
+// 3) The update strategy is rolling
+func isStatefullyStuck(set *apps.StatefulSet, pod *v1.Pod, currentRevision, updateRevision *apps.ControllerRevision) bool {
```

Comment:
I think `isStatefullyStuck` should check whether the replica should be updated.
4) the pod should be updated. `ordinal < int(*currentSet.Spec.UpdateStrategy.RollingUpdate.Partition)`